craft_parts/parts.py

```diff
@@ -56,44 +87,44 @@ class PartSpec(BaseModel):
     build_attributes: List[str] = []
     organize_files: Dict[str, str] = Field({}, alias="organize")
     overlay_files: List[str] = Field(["*"], alias="overlay")
-    stage_files: List[str] = Field(["*"], alias="stage")
-    prime_files: List[str] = Field(["*"], alias="prime")
+    stage_files: List[
+        Annotated[
+            str,
+            AfterValidator(_path_not_empty),
+            AfterValidator(_path_be_relative),
+        ]
+    ] = Field(["*"], alias="stage")
+    prime_files: List[
+        Annotated[
+            str,
+            AfterValidator(_path_not_empty),
+            AfterValidator(_path_be_relative),
+        ]
+    ] = Field(["*"], alias="prime")
```

Comment:
As dicts and lists are mutable, wouldn't `Field(default_factory=dict, alias="organize")`, `Field(default_factory=lambda: ["*"], "alias="overlay")`, ... be more correct here?

Getting some new PyRight validation errors when trying to instantiate a `PartSpec()` that just specifies a plugin, like so:

```python3
    foo = PartSpec(plugin="foo")
```

PyRight shows me:

```
error: Arguments missing for parameters "organize", "overlay", "stage", "prime" (reportCallIssue)
```

Reply:
Since you already proposed a fix in a new PR, I'll resolve this here.